lib/amqp/connection.ex

```diff
@@ -61,91 +50,95 @@ defmodule AMQP.Connection do
+            fail_if_no_peer_cert: true
+          ]
+        )
 
-  ## Connection name
+  ## Backward compatibility for connection name
 
   RabbitMQ supports user-specified connection names since version 3.6.2.
 
-  Connection names are human-readable strings that will be displayed in the management UI.
-  Connection names do not have to be unique and cannot be used as connection identifiers.
-
-  If the name is `:undefined`, the connection is not registered with any name.
-
-  ## Examples
+  Previously AMQP took a connection name as a separate parameter on `open/2` and `open/3` and it is still supported in this version.
 
       iex> options = [host: "localhost", port: 5672, virtual_host: "/", username: "guest", password: "guest"]
       iex> AMQP.Connection.open(options, :undefined)
       {:ok, %AMQP.Connection{}}
 
-      iex> AMQP.Connection.open("amqp://guest:guest@localhost", port: 5673)
+      iex> AMQP.Connection.open("amqp://guest:guest@localhost", "my-connection")
       {:ok, %AMQP.Connection{}}
 
-      iex> AMQP.Connection.open("amqp://guest:guest@localhost", "a-connection-with-a-name")
+      iex> AMQP.Connection.open("amqp://guest:guest@localhost", options, "my-connection")
       {:ok, %AMQP.Connection{}}
 
+  However connection_name parameter is now deprecated and might not be supported in the future versions.
+  You are recommented to pass it with `:connection_name` option instead:
+
+      iex> AMQP.Connection.open("amqp://guest:guest@localhost", connection_name: "my-connection")
+      {:ok, %AMQP.Connection{}}
   """
-  @spec open(keyword | String.t(), String.t() | :undefined | keyword) ::
+  @spec open(String.t() | keyword, keyword | String.t() | :undefined) ::
           {:ok, t()} | {:error, atom()} | {:error, any()}
-  def open(options_or_uri, options_or_name)
+  def open(uri, options)
 
   def open(uri, name) when is_binary(uri) and (is_binary(name) or name == :undefined) do
     open(uri, name, _options = [])
   end
 
   def open(options, name) when is_list(options) and (is_binary(name) or name == :undefined) do
+    {name_from_opts, options} = take_connection_name(options)
+    name = if name == :undefined, do: name_from_opts, else: name
+
     options
     |> merge_options_to_default()
     |> do_open(name)
   end
 
   def open(uri, options) when is_binary(uri) and is_list(options) do
-    open(uri, :undefined, options)
-  end
-
-  @doc """
-  Opens a new connection with a name, merging the given AMQP URI and options.
-
-  This function opens a new connection by merging the given AMQP URI `uri` and
-  the given `options` and assigns it the name `name`.
+    {name, options} = take_connection_name(options)
 
-  The options in `options` take precedence over the values in `uri`.
-
-  See `open/2` for options.
-
-  ## Examples
-
-      iex> AMQP.Connection.open("amqp://guest:guest@localhost", "a-connection-name", port: 5673)
-      {:ok, %AMQP.Connection{}}
+    open(uri, name, options)
+  end
 
-  """
+  @doc false
```

Comment:
You can also use `@deprecated "Use :connection_name in open/2 instead"` or something similar :)

Reply:
ah, nice!